python-package/lightgbm/basic.py

```diff
@@ -555,8 +555,8 @@ def __pred_for_csc(self, csc, num_iteration, predict_type):
 
 class Dataset(object):
     """Dataset in LightGBM."""
-    def __init__(self, data, label=None, max_bin=255, reference=None,
-                 weight=None, group=None, silent=False,
+    def __init__(self, data, label=None, max_bin=None, reference=None,
```

Comment:
May I ask why you've changed `max_bin` to `None` in public interface when it's actually still `255`? I wrote in my previous comment about such cases that they're confusing. If you worry about you'll miss some places while changing the default value in Parameters.rst, just search GitHub with `param=old_value`.

Reply:
it's set by @guolinke

Reply:
@StrikerRUS This is to solve problem of the parameter priority. User can set ```max_bin``` (and other parameters) in both the parameter dict and the function arguments. This will cause the problem: we don't know which one is high "priority" when user set it in both two places. As a result, we can set the default function argument to None. If user didn't set the function argument, we use the value in parameter dict, otherwise use the function arguments. 

We can add something like ```defualt=None (255 in c_api) ``` in python doc, to avoid the addiction search of that parameter.

Reply:
@guolinke 
I suppose it's obvious for users that function argument shouldn't be duplicated in param dict and parameter in signature which has the default value and the own description has higher priority than parameter in dict.

Maybe than it's better to completely remove `max_been` from arguments (for 1 or 2 release mark as deprecated)? Because it's oddly that `None` actually means `255`...

Reply:
some users like to put all parameters into dicts( like me), some like to put them into function arguments. I think set default to None is reasonable, since it is consistent with CLI version.

Reply:
I think we can remove them from arguments as well